### DISCUSSION TITLE
NextJS 15

### DISCUSSION THREAD

**Discussion by imorius** [IdoPesok/zsa#228] [General]
Has anyone experienced Next.js 15 and can tell me if there are any bugs?
_( Translated by ChatGPT =)) )_

**Comment by fhenriquezn** [comment]
The only error I have encountered so far is The "use server" file can only export async functions. Add "async" to the function declaration or return a Promise.
```
export const userSignUpAction = createServerAction()
    .input(userSignUpSchema)
    .handler(async ({ input }: { input: z.infer<typeof userSignUpSchema> }) => {
      // Sleep for .5 seconds
      await new Promise((resolve) => setTimeout(resolve, 500));
      console.log("Input received:", input);
      return input;
    });
```

**Reply by laduniestu** [reply]
Apart from that, there are no problems at all?
